clang/test/CIR/CodeGen/no-pie.c

```diff
@@ -0,0 +1,3 @@
+// RUN: %clang --target=x86_64-unknown-linux-gnu -fclangir %s -fno-PIE -c
+
+void empty(void) {}
```

Comment:
It'd be nice for the test to show the code that's generated for accessing global variables and calling functions with `-fno-PIE` (ideally we can cover all the new conditions that were added).

Reply:
I couldn't find a difference in the produced CIR with `fno-PIE` enabled, or maybe I am wrong? It begs the question of if enabling the flag is enough to add support for this option.

Reply:
Since your change is on `shouldAssumeDSOLocal`, checking whether the output contains or doesn't contain the dso stuff is probably the best thing to do here.

Reply:
updated the test!